Applications/DataHolderLib/ColorLookupTable.h

```diff
@@ -0,0 +1,71 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef COLORLOOKUPTABLE_H
+#define COLORLOOKUPTABLE_H
+
+#include <cassert>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "Applications/DataHolderLib/Color.h"
+
+namespace DataHolderLib
+{
+
+/// Interpolation methods
+enum class LUTType {
+    NONE = 0,
+    LINEAR = 1,
+    EXPONENTIAL = 2,
+    SIGMOID = 3 // not yet implemented
+};
+
+/**
+ * A colour look-up table stored as a vector containing for each entry
+ *    id     - value for which the colour is stored
+ *    colour - RGBA value
+ *    name   - a name referencing the colour (such as a stratigraphic layer)
+ * Colours from the table can then be accessed using either id or name.
+ */
+class ColorLookupTable
+{
+public:
+    ColorLookupTable();
+
+    void setColor(double id, DataHolderLib::Color color);
```

Comment:
Can `color` be passed by `const reference`? The same holds for `setColor()` in line 45.